image/tarexport/save.go

```diff
@@ -205,18 +209,76 @@ func (s *saveSession) save(outStream io.Writer) error {
 			if _, ok := reposLegacy[familiarName]; !ok {
 				reposLegacy[familiarName] = make(map[string]string)
 			}
-			reposLegacy[familiarName][ref.Tag()] = imageDescr.layers[len(imageDescr.layers)-1]
+			reposLegacy[familiarName][ref.Tag()] = imageDescr.layers[len(imageDescr.layers)-1].Encoded()
 			repoTags = append(repoTags, reference.FamiliarString(ref))
 		}
 
+		m := ocispec.Manifest{
+			Versioned: specs.Versioned{
+				SchemaVersion: 2,
+			},
+			MediaType: ocispec.MediaTypeImageManifest,
+			Config: ocispec.Descriptor{
+				MediaType: ocispec.MediaTypeImageConfig,
+				Digest:    digest.Digest(imageDescr.image.ID()),
+				Size:      int64(len(imageDescr.image.RawJSON())),
+				Platform: &ocispec.Platform{
+					OS:           imageDescr.image.OS,
+					Architecture: imageDescr.image.Architecture,
+					Variant:      imageDescr.image.Variant,
+					OSVersion:    imageDescr.image.OSVersion,
+					OSFeatures:   imageDescr.image.OSFeatures,
+				},
```

Comment:
Do we have a `.Features` field that we should also pass through here, even though it's currently unused?  (you know me, seeing platform objects like this always stands out :smile:)

We don't already have a helper somewhere to make a platform object out of these fields, right?

Reply:
We don't have a `.Features` stored on our image object, nor do we seem to have a conversion of *Image to platform.
I'll go ahead and add a function to do this now.